### PR TITLE
test: Introduce a context and a namespace per test and run save tests in parallel

### PR DESCRIPTION
**Pull request by mxinden**
test: Introduce test context per test

test: Clean up e2e test code
* Remove unneeded delete steps
* Improve error feedback
* Cleanup main namespace properly after tests
* Wait for NodePort services to be deleted

test: Run save tests in parallel

### CODE REVIEW

File: test/e2e/alertmanager_test.go [hunk 1]
```diff
@@ -28,86 +28,89 @@ import (
 )
 
 func TestAlertmanagerCreateDeleteCluster(t *testing.T) {
+	t.Parallel()
+
+	ctx := testFramework.NewTestCtx(t)
+	defer ctx.CleanUp(t)
```
Comment (fabxc) [inline test/e2e/alertmanager_test.go]: `CleanUp` -> `Cleanup`. At least we aren't calling it `SetUp` anywhere either.

File: test/e2e/alertmanager_test.go [hunk 2]
```diff
@@ -28,86 +28,89 @@ import (
...
+	ctx.BasicSetup(t, framework.KubeClient)
```
Comment (fabxc) [inline test/e2e/alertmanager_test.go]: This just creates a namespace. In favor of explicitness, I'd just rename this to `ctx.CreateNamespace` and actually have it return the namespace name.

Below we are calling functions expecting a namespace name with `ctx.Id`, which seems somewhat cryptic unless you know that `Id` is always equal to a namespace created during the basic setup.

```go
ns := ctx.CreateNamespace(test) // creates namespace `<ctx.Id>-test`
// ...
framework.DeleteAlertmanagerAndWaitUntilGone(ns, name)
```

This seems a bit clearer overall. What I find important is that it translates seamlessly for tests that need to create multiple namespaces, which would not have an obvious solution from looking at tests that just need `BasicSetup`.

```go
ns1 := ctx.CreateNamespace("test1")
ns2 := ctx.CreateNamespace("test2")
// .. do whatever needed with either namespace
```

File: test/e2e/alertmanager_test.go [hunk 3]
```diff
@@ -28,86 +28,89 @@ import (
 )
 
 func TestAlertmanagerCreateDeleteCluster(t *testing.T) {
+	t.Parallel()
+
+	ctx := testFramework.NewTestCtx(t)
```
Comment (fabxc) [inline test/e2e/alertmanager_test.go]: I think `NewTestCtx` should be a method to the `framework` we are using elsewhere. 
This way you don't have to pass in `framework.KubeClient` if you want this context to create a non-namespaced object.

File: test/e2e/alertmanager_test.go (hunk 2 above)
Comment (fabxc) [inline test/e2e/alertmanager_test.go]: Ah, the Go convention is to always keep abbreviations uppercase, e.g. `ServeHTTP`. So `ctx.Id` should be `ctx.ID`.

**Comment by mxinden** [comment]
Test this please.

File: test/e2e/alertmanager_test.go (hunk 2 above)
Comment (brancz) [inline test/e2e/alertmanager_test.go]: Agreed.

**Comment by brancz** [comment]
Nice work! If it makes things easier maybe close #338 and we'll do everything in here. Your choice @mxinden.

**Comment by mxinden** [comment]
Test this please. CoreOS bot.

**Comment by mxinden** [comment]
test this please.

**Comment by mxinden** [comment]
please test this coreos bot

**Comment by mxinden** [comment]
please test this coreosbot

**Comment by mxinden** [comment]
test this please

**Comment by mxinden** [comment]
please test this

File: test/e2e/alertmanager_test.go (hunk 3 above)
Comment (mxinden) [inline test/e2e/alertmanager_test.go]: @fabxc Ok, I have changed this. Do you want to save the kubeClient in each ctx on creation as well? Would seem strange to me.

**Comment by mxinden** [comment]
@fabxc @brancz Thanks for the feedback. Addressed your comments from #340 and #338 
Any further improvements?

File: test/e2e/alertmanager_test.go [hunk 4]
```diff
@@ -28,86 +28,87 @@ import (
 )
 
 func TestAlertmanagerCreateDeleteCluster(t *testing.T) {
+	t.Parallel()
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+	ns := ctx.CreateNamespaceInCtx(t, framework.KubeClient)
```
Comment (brancz) [inline test/e2e/alertmanager_test.go]: The `InCtx` suffix is unnecessary here I think, as the `CreateNamespace` method would be bound to the ctx struct anyways, that's "context" enough I think.

Comment (fabxc) [inline test/e2e/alertmanager_test.go]: Yep, this.

**Comment by mxinden** [comment]
test this please

**Pull request merged**